src/Parameter/AbstractDateParameter.php

```diff
@@ -2,14 +2,17 @@
 
 namespace CultuurNet\SearchV3\Parameter;
 
+use DateTimeInterface;
+
 abstract class AbstractDateParameter extends AbstractParameter
 {
-    public function getValue()
+    public function __construct(string $dateTime)
     {
-        if ($this->value instanceof \DateTime) {
-            return $this->value->format(\DateTime::ATOM);
-        }
+        $this->value = $dateTime;
+    }
 
-        return $this->value;
+    public static function createFromDateTime(DateTimeInterface $value): self
```

Comment:
Wouldn't it be more logical to flip this around and expect a valid `DateTime` object in the constructor while providing a `fromAtomString()` factory method?

Reply:
From the standpoint of someone actually using this class, yes. 👍 
Was thinking too much from the standpoint of "it needs to become a string".

Reply:
I refactored this in https://github.com/cultuurnet/search-v3/pull/43/commits/60d728a15d21870693101d5a76176838eeebac7a

However we overlooked the use case of wildcards. These are required because SAPI3 has some default (date) filters, to only return events that are relevant at the time of request by default. And to reset those, you need to use a wildcard like `?availableTo=*`

So I had to do a bit of a workaround in a `createWithWildcardValue()` method. Still from a "user" standpoint it's probably clearer this way.

What do you think?

Reply:
Looks good to me. Would even consider simplifying to `DateParameter::wildcard()`

Reply:
Good suggestion, renamed in https://github.com/cultuurnet/search-v3/pull/43/commits/d3b03355f36f99d1aeb9e2d91ed2090b865c5c6b